distributed-tracing/node-js-app/beachshirt/controllers/stylingController.js

```diff
@@ -0,0 +1,57 @@
+/**
+* Driver for styling service which manages different styles of shirts and takes orders for a shirts
+* of a given style.
+* @author Yogesh Prasad Kurmi (ykurmi@vmware.com)
+*/
+const uuid4 = require('uuid4');
+const utils = require('../utils');
+const { Tags, FORMAT_HTTP_HEADERS } = require('opentracing');
+
+module.exports = (app, config, log, tracer) => {
+    app.get('/style', (req, res) => {
+        const parentSpanContext = tracer.extract(FORMAT_HTTP_HEADERS, req.headers)
+        let span = tracer.startSpan('/style', {
+            childOf: parentSpanContext,
+            tags: {[Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_SERVER}
+        });
+        setTimeout(() => {
+            span.setTag(Tags.HTTP_STATUS_CODE, 200)
+            span.finish();
+            res.json(
+                [
+                    {
+                        "name": "Wavefront",
+                        "url": "WavefrontURL"
+                    },
+                    {
+                        "name": "BeachOps",
+                        "url": "BeachOpsURL"
+                    }
+                ]
+            )}, 1000);
+    });
+
+    app.get('/style/:id/make', (req, res) => {
+        const parentSpanContext = tracer.extract(FORMAT_HTTP_HEADERS, req.headers)
+        let span = tracer.startSpan('/style/make', {
+            childOf: parentSpanContext,
+            tags: {[Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_SERVER}
+        });
+        if (Math.floor(Math.random() * 10) === 0) {
+            const error = "Failed to make shirts!";
+            log.error(error);
+            span.setTag(Tags.HTTP_STATUS_CODE, 503)
+            span.setTag(Tags.ERROR, true)
+            span.finish();
+            return res.status(503).json({ error: error });
+        }
+        let shirts = [];
+        for(let i = 0; i < parseInt(req.query.quantity); i++){
+            shirts.push({"name": req.params.id, "imageUrl": `${req.params.id}-image`})
+        }
+        return utils.postRequest(res,
+            `/dispatch/${uuid4()}`,
+            { shirts: JSON.stringify(shirts) },
+            config.delivery, span, tracer)
+    });
+};
```

Comment:
L35: missing semicolon

L36, L48: use const

L46: change to return res.status(503).json({ **error** });

L49: define new const quantity between L48 and L49 and use that const on L49.
        Don't want to parseInt for every iteration.

L50: shirts.push({ name: req.params.id, imageUrl: `${req.params.id}-image` });
        Missing semicolon
        No need to have double quotes on object keys in Javascript unless the keys have special chars.

L55: missing semicolon

Reply:
L48: Should use **const**

L56: Missing semicolon